### ISSUE TITLE
new undo follows have *no* ID

### ISSUE THREAD

**Issue by puckipedia** [tootsuite/mastodon#6211]
```
{
    "id": "https://mastodon.social/users/phschn#follows/1482040/undo",
    "type": "Undo",
    "actor": "https://mastodon.social/users/phschn",
    "object": {
        "type": "Follow",
        "actor": "https://mastodon.social/users/phschn",
        "object": "https://puckipedia.com/"
    },
    "signature": {
        "type": "RsaSignature2017",
        "created": "2018-01-06T18:37:15Z",
        "creator": "https://mastodon.social/users/phschn#main-key",
        "signatureValue": "AuKk5Fsbm4wV6GP2etr0dZLRsyn2AMXXqy56fFo4Wd4OTrPpFurvljsvkeh2mGzGMweoPTP83C/oRC01T4D61EhEo1RAIYKYTE11NeL+Wl8gLPTC6+F5xCQzLCfF00TMBDNL2h2dEnbPsDXjeMrjy1kDSVVmpAFpAnVs3ng6200P5NgvJlpZcBOQbNS/M8OEeGDh/w5y9cmpSRJKDZr3CA6AVihR5FqCvpn9Z6MSCjzR2OKw5rU+Oie8CBkdGwVcJmHBbj68uR3ox6c+smAEcnVM+TVMCzw2GcDkcd0wI4bnUUOSJGjf7fL2P/QFKroANDAfqWe05fPKecZqA7dbzQ=="
    }
}
```
* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by nightpool** [comment]
`https://mastodon.social/users/phschn#follows/1482040/undo`?

**Comment by puckipedia** [comment]
I meant the object inside the Undo

**Comment by nightpool** [comment]
Ah. Well.

**Comment by akihikodaki** [comment]
This is intentional, but since here is a question, we may need a review if it is ActivityPub-conforming.

**Comment by nightpool** [comment]
Note that I don't think this is STRICTLY incorrect, because nested objects can have null ids, indicating that they're anonymous parts of the parent context.

However, all ids have to be dereferenceable and https://mastodon.social/users/phschn#follows/1482040/undo definitely isn't.

**Comment by nightpool** [comment]
(closed because #6221 was merged)

**Comment by puckipedia** [comment]
first, that would mean having a explicitly specified `null` id, and that is for "short lived activities that are not intended to be able to be looked up, such as some kinds of chat messages or game notifications".
Second, you are undoing an existing object. So there's no reason to not give it an ID.
